lib/rspec/core/example_group.rb

```diff
@@ -340,6 +340,28 @@ def self.include_examples(name, *args, &block)
         find_and_eval_shared("examples", name, caller.first, *args, &block)
       end
 
+      # Clear memoized values when adding/removing examples
+      # @private
+      def self.reset_memoized
+        @descendant_filtered_examples = nil
+        @_descendants = nil
+        @parent_groups = nil
+        @declaration_line_numbers = nil
+        @before_context_ivars = nil
```

Comment:
Why do you clear `@before_context_ivars`?  I saw the commit message was "fix memory leak" but I don't understand what the memory leak was.  It's not derived from the list of examples so it doesn't seem like it needs to be cleared.